src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -501,6 +502,14 @@ out providerInstance
 
             Collection<string> stringResult = new Collection<string>();
 
+            // If the directory exists, just return it.
+            // Otherwise fallback to previous behavior and let provider handle it.
+            if (Directory.Exists(userPath))
```

Comment:
can you please explain why having this change? And this might not be the right place for the new code, because a new collection is just allocated above, but you might return without using the collection at all.

Reply:
In this PR I went through so many code ways that I do not remember how I got here.
As far as I see the idea was simple - do not perform an analysis if the path already exists.

Fixed - I return without using the collection.

Reply:
From a quick browsing of the code after this, it seems the new code you added treats the `userPath` as a literal path without consulting `context.SuppressWildcardExpansion`. Say `userPath` is `/blah/a[bc]`, and it happens to have a folder named `a[bc]`.

I'm not that familiar with the file system code, so there could be other concerns on this early return. I suggest to focus on the topic of this PR, and leave optimization to a separate issue/PR.

Reply:
The code was removed.